include/tables/proposals_table.hpp

```diff
@@ -0,0 +1,55 @@
+#include <eosio/eosio.hpp>
+#include <eosio/asset.hpp>
+
+using namespace eosio;
+using std::string;
+
+typedef std::variant<std::monostate, uint64_t, int64_t, double, name, asset, string, bool, eosio::time_point> VariantValue; 
+
+#define DEFINE_PROPOSAL_TABLE TABLE proposal_table { \
+      uint64_t proposal_id; \
+      uint64_t favour; \
+      uint64_t against; \
+      asset staked; \
+      name creator; \
+      string title; \
+      string summary; \
+      string description; \
+      string image; \
+      string url; \
+      time_point created_at; \
+      name status; \
+      name stage; \
+      name type; \
+      uint64_t last_ran_cycle; \
+      uint64_t age; \
+      EOSLIB_SERIALIZE(proposal_table, (proposal_id)(favour)(against)(staked)(creator)(title) \
+        (summary)(description)(image)(url)(created_at)(status)(stage)(type)(last_ran_cycle)(age)) \
+      \
```

Comment:
never seen this statement before, what does it do?

Reply:
That's an statement that tells the compiler that those are the attributes that needs to serialize to get the abi correctly generated. In that table it's needed I can remove it, but it is needed in the referendums_auxiliary table so the variant type is properly generated